u4-databases/features/uni-controller.js

```diff
@@ -0,0 +1,54 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationLists = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iteration = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-iteration-list', { iteration })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const student = (await query).data()
+
+  res.render('uni-student-details', { student })
+}
+
+const courseLists = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const iteration = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-course-list', { iteration })
+}
+
+const courseDetails = async (req, res) => {
+  const code = req.params.code
```

Comment:
ก่อนหน้านี้ในไฟล์ เราใช้ `code` เพื่อแทนโค้ดการวนซ้ำ เช่น `231370-21s1` รหัสนี้แสดงถึงรหัสหลักสูตร เช่น `231370` เพื่อหลีกเลี่ยงความสับสน เราอาจพิจารณาตั้งชื่อตัวแปรให้แม่นยำยิ่งขึ้น: `iterationCode` & `courseCode`

Earlier in the file we used `code` to represent an iteration code, ex: `231370-21s1`. This code represents a course code, ex: `231370`. To avoid confusion, we might consider naming the variables more precisely: `iterationCode` & `courseCode`.